src/recsum/graph_nodes.py

```diff
@@ -0,0 +1,41 @@
+from src.recsum.models import DialogueState
+from src.recsum.response_generator import ResponseGenerator
+from src.recsum.summarizer import Summarizer
+
+
+def update_memory_node(summarizer: Summarizer, state: DialogueState) -> DialogueState:
+    if state.memory is None:
+        state.memory = []
+
+    if state.current_session_index < len(state.dialogue_sessions) - 1:
```

Comment:
this is always `True`, no? you you stop increasing by checking `should_continue_memory_update`

I checked by asserting:
```
    if state.current_session_index >= len(state.dialogue_sessions) - 1:
        raise AssertionError(
            "Current session index is out of bounds for the dialogue sessions."
        )
``` and running main